eZ/Publish/Core/Repository/UserService.php

```diff
@@ -731,40 +732,50 @@ public function updateUser(APIUser $user, UserUpdateStruct $userUpdateStruct): A
-            }
+        $updateUserCallable = function() use ($loadedUser, $userUpdateStruct, $contentService): void {
+            $this->repository->beginTransaction();
+            try {
+                $publishedContent = $loadedUser;
+                if ($userUpdateStruct->contentUpdateStruct !== null) {
+                    $contentDraft = $contentService->createContentDraft($loadedUser->getVersionInfo()->getContentInfo());
+                    $contentDraft = $contentService->updateContent(
+                        $contentDraft->getVersionInfo(),
+                        $userUpdateStruct->contentUpdateStruct
+                    );
+                    $publishedContent = $contentService->publishVersion($contentDraft->getVersionInfo());
+                }
 
-            if ($userUpdateStruct->contentMetadataUpdateStruct !== null) {
-                $contentService->updateContentMetadata(
-                    $publishedContent->getVersionInfo()->getContentInfo(),
-                    $userUpdateStruct->contentMetadataUpdateStruct
+                if ($userUpdateStruct->contentMetadataUpdateStruct !== null) {
+                    $contentService->updateContentMetadata(
+                        $publishedContent->getVersionInfo()->getContentInfo(),
+                        $userUpdateStruct->contentMetadataUpdateStruct
+                    );
+                }
+
+                // User\Handler::update call is currently used to clear cache only
+                $this->userHandler->update(
+                    new SPIUser(
+                        [
+                            'id' => $loadedUser->id,
+                            'login' => $loadedUser->login,
+                            'email' => $userUpdateStruct->email ?: $loadedUser->email,
+                        ]
+                    )
                 );
-            }
 
-            // User\Handler::update call is currently used to clear cache only
-            $this->userHandler->update(
-                new SPIUser(
-                    [
-                        'id' => $loadedUser->id,
-                        'login' => $loadedUser->login,
-                        'email' => $userUpdateStruct->email ?: $loadedUser->email,
-                    ]
-                )
-            );
+                $this->repository->commit();
+            } catch (Exception $e) {
+                $this->repository->rollback();
+                throw $e;
+            }
+        };
 
-            $this->repository->commit();
-        } catch (Exception $e) {
-            $this->repository->rollback();
-            throw $e;
+        if ($isUserProfileUpdateRequested) {
+            // Perform update in user context
+            $updateUserCallable();
+        } else {
+            // Perform update without checking other user permissions
+            $this->repository->sudo($updateUserCallable);
```

Comment:
:worried:

Reply:
If sudo is needed, it would look better imo to specifically check for password change requests, rather than `!$isUserProfileUpdateRequested`. In case there are, or could later be, more cases than these two (content edit and password change).

Reply:
:heavy_check_mark: 

We've discussed this and password change will probably be separated from the user update in general in this service, to prevent accidently introducing security issues.